src/modules/commander/HealthAndArmingChecks/checks/estimatorCheck.cpp

```diff
@@ -669,6 +671,41 @@ void EstimatorChecks::gpsNoLongerValid(const Context &context, Report &reporter)
 	}
 }
 
+void EstimatorChecks::lowPositionAccuracy(const Context &context, Report &reporter,
+		const vehicle_local_position_s &lpos) const
+{
+	switch (reporter.failsafeFlags().local_position_accuracy_low) {
+	case true:
+
+		// check conditions for re-setting flag
+		if (reporter.failsafeFlags().local_position_invalid || !(_param_com_low_eph.get() > FLT_EPSILON
+				&& lpos.eph > _param_com_low_eph.get())) {
+			reporter.failsafeFlags().local_position_accuracy_low = true;
+		}
+
+		break;
+
+	default:
+
+		// check conditions for setting flag
+		if (context.isArmed() && !reporter.failsafeFlags().local_position_invalid && (_param_com_low_eph.get() > FLT_EPSILON
+				&& lpos.eph > _param_com_low_eph.get())) {
+			reporter.failsafeFlags().local_position_accuracy_low = true;
```

Comment:
Can you simplify this to:
```
bool local_position_accuracy_low = ...;
if (!reporter.failsafeFlags().local_position_accuracy_low && local_position_accuracy_low) {
  [report]
}
reporter.failsafeFlags().local_position_accuracy_low = local_position_accuracy_low;
```

Reply:
fix incorporated in 8188de419a507df446cc34f94cf9e8bb0d211028

Reply:
I now always set the flag, but only report and take action if armed.